### ISSUE TITLE
Conjugate resetting after loading a model

### ISSUE THREAD

**Issue by LeviLingsch** [pytorch/pytorch#81690]
If you have a question or would like help and support, please ask at our
[forums](https://discuss.pytorch.org/).

If you are submitting a feature request, please preface the title with [feature request].
If you are submitting a bug report, please fill in the following details.

## Issue description

Using a matrix and its conjugate within a model poses no problems when using the `torch.conj()` method. The problem arises when the model is loaded in. Although we have access to the conjugate matrix within the model, it is not actually the conjugate of the matrix. 

## Code example
`

    import torch
    class Test (torch.nn.Module):
    def __init__(self):
        self.m = torch.randn(10,10, dtype=torch.cfloat)
    def forward(self):
        self.V = torch.randn(10,10, dtype=torch.cfloat)
        self.V_ct = torch.conj(self.V)
    t = Test()
    t.forward()
    print(t.V_ct)
    torch.save(t, "test.pth")
    t1 = torch.load("test.pth")
    print(t1.V_ct)
`

## System Info

PyTorch version: 1.11.0
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.3 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.9.7 | packaged by conda-forge | (default, Sep 29 2021, 19:20:46)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.13.0-51-generic-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: 11.2.152
GPU models and configuration: 
GPU 0: NVIDIA GeForce RTX 3090
GPU 1: NVIDIA GeForce RTX 3090
GPU 2: NVIDIA GeForce RTX 3090
GPU 3: NVIDIA GeForce RTX 3090

Nvidia driver version: 470.129.06
cuDNN version: Probably one of the following:
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn.so.8
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_adv_train.so.8
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_ops_train.so.8
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.21.2
[pip3] pytorch-metric-learning==1.3.0
[pip3] torch==1.11.0
[pip3] torchaudio==0.11.0
[pip3] torchvision==0.12.0
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               11.3.1               h2bc3f7f_2  
[conda] libblas                   3.9.0            12_linux64_mkl    conda-forge
[conda] libcblas                  3.9.0            12_linux64_mkl    conda-forge
[conda] liblapack                 3.9.0            12_linux64_mkl    conda-forge
[conda] liblapacke                3.9.0            12_linux64_mkl    conda-forge
[conda] mkl                       2021.4.0           h06a4308_640  
[conda] mkl-service               2.4.0            py39h7f8727e_0  
[conda] mkl_fft                   1.3.1            py39hd3c417c_0  
[conda] mkl_random                1.2.2            py39h51133e4_0  
[conda] numpy                     1.21.2           py39h20f2e39_0  
[conda] numpy-base                1.21.2           py39h79a1101_0  
[conda] pytorch                   1.11.0          py3.9_cuda11.3_cudnn8.2.0_0    pytorch
[conda] pytorch-metric-learning   1.3.0                    pypi_0    pypi
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] torch                     1.8.1+cu101              pypi_0    pypi
[conda] torchaudio                0.8.1                    pypi_0    pypi
[conda] torchvision               0.9.1+cu101              pypi_0    pypi

cc @ezyang @gchanan @zou3519 @anjali411 @dylanbespalko @mruberry @Lezcano @nikitaved

**Comment by LeviLingsch** [comment]
I have also found that using `torch.resolve_conj()` was able to resolve the issue in the one case I tested. I will use this in the meantime and hope that nothing else pops up. 

Thanks for the help, @anjali411.

**Comment by anjali411** [comment]
@LeviLingsch yup `torch.resolve_conj()` will give you a tensor with conjugate bit set to 0 (resolved conjugation) so that sounds good too

**Comment by ezyang** [comment]
We still need to fix this by ensuring conjugate bit gets serialized (or we resolve upon serialization, but we preserve sharing on serialization so putting in the conj bit is better.

**Comment by vadimkantorov** [comment]
@ezyang Regarding very other math bits, does pytorch has an easy way to get/set CPU math bits like fast math? (loading 3rd party shared libraries may change it and then cause non-reporducibility)

**Comment by ezyang** [comment]
This would be a simple enough utility to add, happy to review a PR for it.

**Comment by kshitij12345** [comment]
> @ezyang Regarding very other math bits, does pytorch has an easy way to get/set CPU math bits like fast math? (loading 3rd party shared libraries may change it if they were compiled with fastmath and then cause slightly changed behavior of CPU math) i know there is a knob for denorms, but are there any other bits? (might be good to have a way to get/set all of them in one go)

IIUC, I think this should be a separate issue (as PR fixing this won't address these knobs). wdyt @ezyang?

**Comment by ezyang** [comment]
yes

**Comment by vadimkantorov** [comment]
> @vadimkantorov you might want to open a different issue for the above request :)

Done! https://github.com/pytorch/pytorch/issues/88805
